src/lib/expansion-panel/expansion-panel.ts

```diff
@@ -0,0 +1,142 @@
+import {
+  Component,
+  Directive,
+  EventEmitter,
+  Input,
+  Output,
+  ViewEncapsulation,
+  AfterContentInit,
+  OnDestroy,
+  ContentChildren,
+  QueryList,
+  Optional,
+  forwardRef,
+} from '@angular/core';
+import {
+  trigger,
+  state,
+  style,
+  transition,
+  animate,
+} from '@angular/animations';
+import {Subscription} from 'rxjs/Subscription';
+
+/** Used to generate unique ID for each expansion panel. */
+let nextId = 0;
+/** Time and timing curve for expansion panel animations. */
+export const EXPANSION_PANEL_ANIMATION_TIMING = '225ms cubic-bezier(0.4,0.0,0.2,1)';
+
+@Directive({
+  selector: 'md-expansion-panel-set, mat-expansion-panel-set',
```

Comment:
A group of expansion panels is typically called an accordion. Accordions also have special a11y treatment on top of the standalone panel: https://www.w3.org/TR/wai-aria-practices-1.1/#accordion. It's possible that this maybe be a completely separate component that just shares some code with the expansion panel.

Maybe leave this out for now and add it in a follow-up?